### ISSUE TITLE
How to connect physical midi keyboard in windows

### ISSUE THREAD

**Issue by ericfont** [zynaddsubfx/zyn-fusion-issues#119]
I'm a long time linux user of zynaddsubfx...but I can't figure out how to connect my midi controller to Zyn-Fusion in Windows.

**Comment by fundamental** [comment]
On Wed, Oct 25, 2017 at 02:30:09AM -0700, Eric Fontaine wrote:
> I'm a long time linux user of zynaddsubfx...but I can't figure out how to connect my midi controller to Zyn-Fusion in Windows.

I'm assuming you're talking about the standalone.
The windows standalone should attempt to connect to all midi controllers it can
find at startup.

Zyn does not attempt to connect to any midi devices plugged in after Zyn has
started up.

**Comment by ericfont** [comment]
Turns out I had started Zyn-Fusion from the executable named zyn-fusion, and nothing worked at all because I realized that binary was solely for the UI.  When I started from zynaddsubfx executable, then that apparently started the backend and then launched the frontend UI.

I'm still not hearing audio output as I mentioned here: #120
I'm also noticing that pressing notes on my midi controller (M-Audio Keystation 49e) will cause the notes in zyn-fusion's UI to flash, but I don't see the audio amplitude meter respond at all...only when I click notes on the midi keyboard do I see that meter respond.  I'm also noting that the Octave -/+ buttons on my midi controller seem to be interpreted to zyn to adjust up/down chromatically instead of up/down octaves.

**Comment by fundamental** [comment]
Are you sure that you have the right MIDI channel coming from your physical controller?

You can change which one zyn is listening to via "part settings"->"instrument settings"->"active midi channel" (bottom left of the sub-panel)

As per the octave -/+ issue, that sounds odd. My MIDI keyboard simply changes the notes that it emits after hitting the octave -/+ buttons and never produces any event which a software synth could receive.

**Comment by ericfont** [comment]
Thanks...I had not realized windows had put my midi controller into a channel different from the default CH1, and after trying a few channels, I found the correct one.

**Comment by ericfont** [comment]
As per the octave -/+ issue, after a bit of investigating, it turns out that your code isn't at fault...I found I had the same problem in musescore too...and it turns out I pressed some combination of keys on my m-audio keystation49e which changed the octave -/+ into chromatic -/+, and after resetting it to octave, that works.

Thanks, and sorry for wasting your time, but your response helped me realize it probably wasn't your program at fault.  I'm closing this issue as it has been resolved.
